<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

	<!--arg name="ik_file" 	default="/srv/data/gait1992/inverse_kinematics/task_InverseKinematics.mot" />
	<arg name="model_file" 	default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" / -->
	<!-- arg name="ik_file" 	default="/catkin_ws/src/osrt_ros/data/Gait6_ik_clean.mot" /-->
	<arg name="ik_file" 	default="/catkin_ws/src/osrt_ros/data/Gait6_ik.mot" />
	<arg name="model_file" 	default="/catkin_ws/src/osrt_ros/data/FK scaled model.osim" />
	<arg name="rate_divider" default="1" />


        <node pkg="opensimrt_bridge" type="ik_dump" 	name="inverse_kinematics_from_file" output="screen">
		<param name="ik_file" 			value="$(arg ik_file)" />
                <param name="model_file" 		value="$(arg model_file)" />
                <param name="remove_n_last_rows" 	value="0" />
		<param name="simulation_loops" 		value="1" />
		
		<param name="start_at_secs" 		value="1668695814" />
		<param name="start_at_nsecs" 		value="643890142" />
		<param name="stop_at_secs" 		value="1668695816" />
		<param name="stop_at_nsecs" 		value="883890142" />

                <param name="resample_period" 		value="0.01" />
		<!-- maybe you want to run this node slower than the resampling period to give a lot of time between messages for the other algorithms to run. Otherwise this number should be one -->
		<param name="rate_divider" 		value="$(arg rate_divider)" />

		<remap from="/inverse_kinematics_from_file/r_data" to="/ik/output"/>
		<remap from="/inverse_kinematics_from_file/out_labels" to="/ik/outlabels"/>

        </node>

</launch>
